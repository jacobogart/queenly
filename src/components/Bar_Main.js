import React, { Component } from 'react';

// * CSS imports
import '../css/Main_Info.css';

export default class Bar_Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  render() {
    let { name, imageURL, phone, notes, website, facebook, instagram, twitter } 
      = this.props.cardData;
    let { webIcon, phoneIcon, instaIcon, facebookIcon, twitterIcon}
      = this.props.icons;

    return (
      <main className="Bar-Main Main_Info">
        <article className="Content">
          <h2>{ name }</h2>
          <img className="bar-img" src={ imageURL } alt={"Image of " + { name }} />
          <p className="phone">
            { phoneIcon }
            { phone }
          </p>
          <h4>Notes:</h4>
          <p className="notes">{ notes }</p>
        </article>
        <aside className="social-links">
          <a href={ website }>
            { webIcon }
          </a>
          <a href={ instagram }>
            { instaIcon }
          </a>
          <a href={ facebook }>
            { facebookIcon }
          </a>
          <a href={ twitter }>
            { twitterIcon }
          </a>
        </aside>
      </main>
    );
  }
}