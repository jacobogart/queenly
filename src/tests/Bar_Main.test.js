import React from 'react';
import ReactDOM from 'react-dom';
import Bar_Main from '../components/Bar_Main.js';
import { shallow } from 'enzyme';

const [nameOfFunctionHere] = jest.fn();

describe('Bar_Main', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Bar_Main />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
// Put state here
    });
  });
})