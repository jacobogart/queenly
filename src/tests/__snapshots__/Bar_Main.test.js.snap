// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar_Main should match snapshop 1`] = `
<main
  className="Bar-Main Main_Info"
>
  <article
    className="Content"
  >
    <h2>
      Meeting House Tavern
    </h2>
    <img
      alt="Image of [object Object]"
      className="bar-img"
      src="https://scontent-dfw5-1.cdninstagram.com/vp/dc393598c8355083e81b210447749477/5D30B2CA/t51.2885-19/s320x320/30920559_218883882204264_3986740088989024256_n.jpg?_nc_ht=scontent-dfw5-1.cdninstagram.com"
    />
    <p
      className="phone"
    >
      (773) 696-4211
    </p>
  </article>
  <aside
    className="social-links"
  >
    <a
      href="http://www.meetinghousetavern.com/"
    />
    <a
      href="https://www.instagram.com/meetinghousetavernchi/"
    />
    <a
      href="https://www.facebook.com/MeetingHouseChi"
    />
    <a
      href="https://twitter.com/meetinghousechi"
    />
  </aside>
</main>
`;
